src/components/custom-aggrid/custom-aggrid-header.js

```diff
@@ -73,11 +135,11 @@ const CustomHeaderComponent = ({
         setIsHoveringHeader(false);
     };
 
-    const isFilterOpened = Boolean(filterAnchorEl);
-    const popoverId = isFilterOpened ? `${field}-filter-popover` : undefined;
-    const isFilterActive = !!filterOptions.length;
-    const isFilterIconDisplayed =
-        isHoveringHeader || !!filterSelectedOptions.length || isFilterOpened;
+    useEffect(() => {
+        if (!filterSelector) {
+            setSelectedFilterData(undefined);
+        }
+    }, [filterSelector]);
```

Comment:
This useEffect seems to be useless

Reply:
if filterSelector is changed to null, that means:
 - there is no selected filter or 
 - nmk type was changed or 
 - tab was changed
 we are using this useEffect using filterSelector because the state of selectedData is managed internally in this component to improve performance